plugins/modules/identity/keycloak/keycloak_client_scopemapping.py

```diff
@@ -0,0 +1,333 @@
+        description:
+            - State of the client-level role scope mapping.
+            - On C(present), the client-level role scope mapping will be created if it does not yet exist, or updated with the parameters you provide.
+            - On C(absent), the client-level role scope mapping will be removed if it exists.
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+    realm:
+        type: str
+        description:
+            - They Keycloak realm under which this role representation resides.
+        default: 'master'
+    client_id:
+        type: str
+        description:
+            - Name of the client to be mapped (different than I(cid)).
+            - This parameter is required (can be replaced by cid for less API call).
+    id:
+        type: str
+        description:
+            - ID of the client to be mapped.
+            - This parameter is not required for updating or deleting the rolemapping but
+              providing it will reduce the number of API calls required.
+    client_role:
+        description: Scope mapping of the client-level role.
+        type: dict
+        suboptions:
+            id:
+                description: ID of the client
+                type: str
+            client_id:
+                description: Name of the client
+                type: str
+            roles:
+                description: Client roles
+                type: list
+                elements: dict
+                suboptions:
+                    name:
+                        description: Name of the role
+                        type: str
+                    id:
+                        description: ID of the role
+                        type: str
+
+extends_documentation_fragment:
+- community.general.keycloak
+
+
+author:
+    - Fynn Chen (@fynncfchen)
+'''
+
+EXAMPLES = '''
+- name: Map a client role to a client, authentication with credentials
+  community.general.keycloak_client_scope_mapping:
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    state: present
+    client_id: client1
+    client_role:
+        client_id: client2
+        roles:
+          - name: role_name1
+            id: role_id1
+          - name: role_name2
+            id: role_id2
+  delegate_to: localhost
+
+- name: Map a client role to a client, authentication with token
+  community.general.keycloak_client_scope_mapping:
+    realm: MyCustomRealm
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    token: TOKEN
+    state: present
+    id: 6e1d65de-f01a-4d3a-b7c6-3d581165966f
+    client_role:
+        id: 6e1d65de-f01a-4d3a-b7c6-3d581165966f
+        roles:
+          - name: role_name1
+            id: role_id1
+          - name: role_name2
+            id: role_id2
+  delegate_to: localhost
```

Comment:
The name of this example implies that it is the same example as the former one, but with different authentication solution.  However, I note that the client-id's are different - which may be confusing.

However, I'm not actually sure of the benefit to the reader by showing both authentication variations, particularly as there's no example included of how to get the token itself in ansible.

Reply:
The clients' IDs should be different, one is for the client to be granted, another is the client with roles, I know it's confusing but I have no idea for naming them differently, for now.

Reply:
Hi @fynncfchen.  My concern wasn't with there being different client Id's, but rather than the name of the examples compared to the name of the previous example implied there was only a difference in how the module authenticates.